File: src/main/java/com/tobe/healthy/lessonhistory/domain/dto/in/RetrieveLessonHistoryByDateCond.java
package com.tobe.healthy.lessonhistory.domain.dto.in;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.Locale;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "조회 조건 DTO")
public class RetrieveLessonHistoryByDateCond {

    @Schema(description = "조회 날짜", example = "YYYY-MM", required = false)
    @Builder.Default
    private String searchDate = formatDate(LocalDate.now());

    private static String formatDate(LocalDate lessonDt) {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("YYYY-MM", Locale.KOREAN);
        return lessonDt.format(formatter);
    }
}

File: src/main/kotlin/com/tobe/healthy/lessonhistory/domain/dto/in/RetrieveLessonHistoryByDateCond.kt
package com.tobe.healthy.lessonhistory.domain.dto.`in`

import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDate
import java.time.LocalDate.now
import java.time.format.DateTimeFormatter
import java.util.Locale.KOREAN

@Schema(description = "조회 조건 DTO")
data class RetrieveLessonHistoryByDateCond(
    @Schema(description = "조회 날짜", example = "YYYY-MM", required = false)
    val searchDate: String = formatDate(now())
) {
    companion object {
        private fun formatDate(lessonDt: LocalDate): String {
            val formatter = DateTimeFormatter.ofPattern("YYYY-MM", KOREAN)
            return lessonDt.format(formatter)
        }
    }
}